libraries/AP_HAL/system.h

```diff
@@ -28,4 +28,48 @@ uint64_t native_millis64();
 void dump_stack_trace();
 void dump_core_file();
 
+template <typename T>
+inline bool timeout_expired(const T past_time, const T now, const T timeout)
+{
+    const T dt = now - past_time;
+    return (dt > timeout);
+}
+
+template <typename T>
+inline T timeout_remaining(const T past_time, const T now, const T timeout)
+{
+    const T dt = now - past_time;
```

Comment:
type assert unsigned int? or just make it uint32_t only

Reply:
Added type assert